File: 0230_Kth Smallest Element in a BST/KthSmallestElementinaBST.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {

    private int val = -1;
    private int searchCount = 0;

    public int KthSmallest(TreeNode root, int k){
        searchCount = k;
        Search(root);
        return val;
    }

    private void Search(TreeNode node){
        if (node == null || searchCount == 0) return;
        Search (node.left);

        searchCount--;
        if (searchCount == 0) {
            val = node.val;
            return;
        }
        Search(node.right);
    }
}

File: 0230_Kth Smallest Element in a BST/KthSmallestElementinaBST.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def kthSmallest(self, root, k):
        """
        :type root: TreeNode
        :type k: int
        :rtype: int
        """
        if not root or k<=0:return 0
        count = 0
        stack = []
        while root or stack:
            while root:
                stack.append(root)
                root = root.left
            root = stack.pop()
            count+=1
            if count == k:return root.val
            root = root.right
        return 0